paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/ColumnPruningAndPushDown.scala

```diff
@@ -82,12 +92,14 @@ trait ColumnPruningAndPushDown extends Scan with Logging {
 
   lazy val readBuilder: ReadBuilder = {
     val _readBuilder = table.newReadBuilder().withReadType(readTableRowType)
-    if (filters.nonEmpty) {
-      val pushedPredicate = PredicateBuilder.and(filters: _*)
-      _readBuilder.withFilter(pushedPredicate)
+    if (pushedPartitionFilters.nonEmpty) {
+      _readBuilder.withPartitionFilter(new AndPartitionPredicate(pushedPartitionFilters.asJava))
```

Comment:
Do not wrap to `AndPartitionPredicate`, if size is 1, we can introduce `PartitionPredicate.and` static method.